pinot-controller/src/main/java/org/apache/pinot/controller/api/resources/PinotSegmentRestletResource.java

```diff
@@ -901,6 +903,37 @@ public String getServerMetadata(
     return segmentsMetadata;
   }
 
+  @GET
+  @Path("segments/{tableName}/zkmetadata")
+  @Produces(MediaType.APPLICATION_JSON)
+  @ApiOperation(value = "Get the zookeeper metadata for all table segments",
+          notes = "Get the zookeeper metadata for all table segments")
+  public String getZookeeperMetadata(
+          @ApiParam(value = "Name of the table", required = true) @PathParam("tableName") String tableName,
+          @ApiParam(value = "OFFLINE|REALTIME") @QueryParam("type") String tableTypeStr)
+          throws JsonProcessingException {
+    LOGGER.info("Received a request to fetch zookeeper metadata for all segments for table {}", tableName);
+    TableType tableType = Constants.validateTableType(tableTypeStr);
+
+    List<String> tableNamesWithType =
+            ResourceUtils.getExistingTableNamesWithType(_pinotHelixResourceManager, tableName, tableType, LOGGER);
+    Map<String, JsonNode> tableTypeToSegmentMetadata = new HashMap<>();
+    for (String tableNameWithType: tableNamesWithType) {
+      List<String> segments =
+              _pinotHelixResourceManager.getSegmentsFor(tableNameWithType, false);
```

Comment:
You may use `getSegmentsZKMetadata()` to batch read all ZK metadata for a table, which is more efficient